app/Http/Controllers/CanteensController.php

```diff
@@ -46,10 +44,41 @@ public function byWeek(Request $request)
         
         $canteens = Canteens::with('menus')->whereBetween('date', [$start,$end])->orderBy('date')->get();
         
-        if ($canteens->isEmpty()) {
+        
+        
+        return self::_after($request, $canteens);
+    }
+    
+    
+    private static function _modSince($request)
+    {
+        return new Carbon(str_replace(':', '', $request->header('If-modified-since', ': 1970-01-01')));
```

Comment:
Could you explain it why was it necessary?

Reply:
For some weird reason, the `$request->header` line returns a line like `: Mon, 02 Jul 2018 00:00:00`. Notice the __:__ - it's an unexpected char and `Carbon` can not parse it. So we remove it with `str_replace`.
I've tried to use header names with the colon, but it always returns `null`, as it's not part of the header name.